### PR TITLE
Added http headers in opentelemetry span  logs table

### PR DESCRIPTION
**Pull request by jonymohajanGmail**
### Changelog category (leave one):

- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Added HTTP headers to OpenTelemetry span logs table for enhanced traceability.

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

File: src/Server/HTTPHandler.cpp [hunk 1]
```diff
@@ -1060,6 +1060,9 @@ void HTTPHandler::handleRequest(HTTPServerRequest & request, HTTPServerResponse
             context->getOpenTelemetrySpanLog());
         thread_trace_context->root_span.kind = OpenTelemetry::SERVER;
         thread_trace_context->root_span.addAttribute("clickhouse.uri", request.getURI());
+        thread_trace_context->root_span.addAttribute("http_referer",  session->getClientInfo().http_referer);
```
Comment (FrankChen021) [inline src/Server/HTTPHandler.cpp:1063]: for these attribute names, I'm suggesting to follow the newest OpenTelemetry standard attribute names for http server spans.

https://opentelemetry.io/docs/specs/semconv/http/http-spans/#http-server-definitions

Comment (jonymohajanGmail) [reply src/Server/HTTPHandler.cpp]: Fixed. Thanks for the suggestion.

**Comment by robot-ch-test-poll2** [comment]
*This is an automated comment for commit de39b79a785af50d6c3ef47502e6a7cd8dc7635b with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/70516/de39b79a785af50d6c3ef47502e6a7cd8dc7635b/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/df4e80bd01c4baaba95cd9471a7f6b95ac5db465/docs_check.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

**Comment by jonymohajanGmail** [comment]
@alexey-milovidov Could you please assist with merging this pull request?

**Review by alexey-milovidov — COMMENTED** [review]
Looks good, but needs a test, at least for HTTP and TCP.

**Review by alexey-milovidov — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
Thank you so much, this is perfect!

**Pull request merged**